File: ELEC881_Tabnine/src/No Description Prompt/Java script (JS)/q18-count-numbers-with-unique-digits.js
/**
 * @param {number} n
 * @return {number}
 */
var countNumbersWithUniqueDigits = function(n) {
    if (n === 0) return 1;
    let count = 10;
    let availableDigits = 9;
    for (let i = 1; i < n; i++) {
        availableDigits *= (9 - i + 1);
        count += availableDigits;
    }
    return count;
};

File: ELEC881_Tabnine/src/Original Prompt/Python/q18-count-numbers-with-unique-digits.py
'''
Given an integer n, return the count of all numbers with unique digits, x, where 0 <= x < 10n.

 

Example 1:

Input: n = 2
Output: 91
Explanation: The answer should be the total numbers in the range of 0 ≤ x < 100, excluding 11,22,33,44,55,66,77,88,99
Example 2:

Input: n = 0
Output: 1
 

Constraints:

0 <= n <= 8
'''
class Solution(object):
    def countNumbersWithUniqueDigits(self, n):
        """
        :type n: int
        :rtype: int
        """
        dp = [1] * (n + 1)
        for i in range(1, n + 1):
            dp[i] = dp[i - 1] * (10 - i + 1)
        return sum(dp)
    
    